README.md

```diff
@@ -206,3 +206,247 @@ Check logs across pods using `kc_logs.sh` script from
 ```bash
 kc_logs.sh razee razeedash-api 1m
 ```
+
+## Web hooks
+
+Implemented web hooks so data in Razeedash can be augmented by third-party
+services such as test suites and vulnerability scanners.
+
+Components are:
+
+- Web hook creation
+- Web hook deletion
+- Trigger by resource ID
+- Trigger by Regex
+- Callback response
+
+### Web hook definition
+
+`POST /v2/webhook/` will create a web hook and the header should contain the `razee-org-key`
+
+JSON body for web hook triggered by cluster change:
+
+```json
+{
+          "cluster_id": "ID of the cluster",
+          "trigger": "cluster",
+          "kind": "Deployment",
+          "field": "searchableData.name",
+          "filter": "regex string example, `(watch-keeper)`",
+          "service_url": "URL of service to POST upon triggering"
+}
+```
+
+**Note: field and filter are optional. If not defined the above will fire if a
+`deployment` resource kind is changed**
+
+JSON body for web hook triggered by image change:
+
+```json
+{
+          "trigger": "image",
+          "kind": "image",
+          "field": "name",
+          "filter": "regex string example, `(quay.io\\/mynamespace)`",
+          "service_url": "URL of service to POST upon triggering"
+}
+```
+
+**Note:** field and filter are optional. If not defined the above will fire if
+any new image is deployed on an organization's clusters
+
+- kind: Kind would be the type of resource or in a special case, images
+- trigger: cluster, image
+- id: ID of the resource used in trigger OR
+- field: field dot-notation into JSON to apply a filter
+- filter: regex parameter to filter if the trigger should fire or not if field
```

Comment:
Does webhook trigger whenever field matches the filter, or does filter omit any change where field matches filter?

Reply:
Matches

Reply:
Updated docs